### DISCUSSION TITLE
There are some Duplicate column between  Foreign key and Navigation Property at locode, How to controll it just show one of them?

### DISCUSSION THREAD

**Discussion by 1257960069** [ServiceStack/Discuss#113] [General]
![image](https://github.com/ServiceStack/Discuss/assets/31265244/2c9c12aa-d333-42fc-9275-f041fc3ee058)
![image](https://github.com/ServiceStack/Discuss/assets/31265244/766d9646-fd38-4e8d-9b40-dc59f15e3e07)

**Comment by 1257960069** [comment]
![image](https://github.com/ServiceStack/Discuss/assets/31265244/62fc9e11-b80e-44b6-818e-97eee4708ee1)
can we show name, ranther  than foreign key id by default?  if id add the Navigation Property, it will show 2 column.

**Comment by Layoric** [comment]
@1257960069 You can use the attribute `[Format(FormatMethods.Hidden)]` on the properties to handle the duplicates. You can take a look at how this was handled in our [example app TalentBlazor](https://github.com/NetCoreApps/TalentBlazor/blob/main/TalentBlazor.ServiceModel/Talent.cs#L79).

For changing how the FK is used, you can use the [Ref](https://docs.servicestack.net/locode/programmatic#custom-ui-references) attribute, eg 

```csharp
public class Employee
{
    public int Id { get; set; }
    public string FirstName { get; set; }
    public string LastName { get; set; }

    [Ref(Model = nameof(Employee), RefId = nameof(Id), RefLabel = nameof(LastName))]
    public int ReportsTo { get; set; }
    //...
}
```

Hope that helps.

**Reply by 1257960069** [reply]
`[Ref(Model = nameof(Employee), RefId = nameof(Id), RefLabel = nameof(LastName))]`
 just work at creating and editing, not work at table column

**Reply by 1257960069** [reply]
thanks, it worked  with `[Format(FormatMethods.Hidden)]` to hide Navigation Property

**Reply by 1257960069** [reply]
![image](https://github.com/ServiceStack/Discuss/assets/31265244/d50472a0-a027-412e-ac25-4556f302a80f)
When filtering a foreign key column, can we show dialog to select a id, just like creating select foreign  key id?

**Reply by mythz** [reply]
Please submit any feature requests to https://servicestack.net/ideas
